docker/bookkeeper/entrypoint.sh

```diff
@@ -74,6 +73,28 @@ create_bookie_dirs() {
   done
 }
 
+# Create a Bookie ID if this is a newly added bookkeeper pod
+# or read the Bookie ID if it already exists
+set_bookieid() {
+  IFS=',' read -ra directories <<< $1
+  BK_ID_DIR="${directories[0]}"
+  BK_ID_FILE="${BK_ID_DIR}/id"
```

Comment:
I'm not sure why we are reading this `id` file. From the BP, it is present only in the configuration file:

> This BookieId will be present only in the configuration file and it is the key to lookup the Cookie on the MetadataService. Inadvertently changing the BookieId will prevent the Bookie to boot as it won’t find a matching Cookie. There is no need to store the BookieId on the cookie or persist it on the local storage (ledgers, indexes or journal directories).

https://bookkeeper.apache.org/bps/BP-41-bookieid/

Reply:
@fpj the reason for persisting the bookieId within the `id` file was because in case of a bookie getting restarted, we would not want to set a new value for the bookieId, instead we would want to use the same value of bookieId we were using before the pod got restarted and expose this value as an environment variable within the pod, and the only way to read the older value would be by persisting this value.

Reply:
@fpj in order projects, like HerdDB, we had the same approach, to store the bookieId in a file on the same directory.
assuming that the bookieid is strictly bound to the disks (contents) and not to the network address

Reply:
This code is only writing the bookie id to a single directory (`BK_ID_DIR="${journal_directories[0]}"
  BK_ID_FILE="${BK_ID_DIR}/id"`). If we are trying to tie the bookie id to the disks (content), then do we have to write it to all directories and check accordingly?

Reply:
What would startup (entrypoint) do if bk id is different across directories, which could be possible if restarts are frequent.

Reply:
> then do we have to write it to all directories and check accordingly?
This is a good point. I agree

> What would startup (entrypoint) do if bk id is different across directories, which could be possible if restarts are frequent.
IMHO we should fail the startup and let some sysadmin to deal with the problem. Probably we mounted wrong disks